client/analytics/report/taxes/index.js

```diff
@@ -20,22 +21,46 @@ import ReportSummary from 'analytics/components/report-summary';
 import TaxesReportTable from './table';
 
 export default class TaxesReport extends Component {
+	getChartMeta() {
+		const { query } = this.props;
+		const isCompareTaxView = 'compare-tax-codes' === query.filter;
```

Comment:
I think we should check `query.taxes.split( ',' ).length` here too. Otherwise, the mode immediately switches to compare mode when you change the filter, instead of after clicking the 'compare' button.